### DISCUSSION TITLE
max-lines rule

### DISCUSSION THREAD

**Discussion by dagmon2** [golangci/golangci-lint#2881] [Linter Ideas] [answered]
Hello!

i was wondering if currently there is a support for a rule
that enforces maximum lines per file

similar to this eslint rule - https://eslint.org/docs/rules/max-lines

Thank you very much!
dagmon.

**Comment by finalight** [comment]
need this as well

**Comment by hendrywiranto** [comment]
Hi, I think you should just submit an issue since it has more visibility by the maintainers
also this may be a new linter idea for golangci

**Comment by herzadinata** [comment]
will really appreciate it if it's added

**Reply by ccoVeille** [reply]
The only linter I know that is somehow providing something close to this is 

revive.function-length

https://revive.run/r#function-length

Available via revive linters in golangci-lint

https://golangci-lint.run/usage/linters/#revive

You could open a PR on this project maybe

**Reply by herzadinata** [reply]
Actually golangci-lint has already included `funlen` linter to limit lines in a function (yes I'm using it).
But currently golangci-lint doesn't have linter to limit lines in an entire .go file.

**Reply by ccoVeille** [reply]
Indeed. But revive is a good candidate if you want to add this to a linter

**Reply by ccoVeille** [reply]
The `filen` linter has been added, it's great.

But for archive and anyone interested, there  is a pending PR about adding the same feature into revive rules
 https://github.com/mgechev/revive/pull/1072#issuecomment-2437009067

**Comment by ldez** [comment, marked as answer]
https://github.com/golangci/golangci-lint/pull/5081

**Reply by ccoVeille** [reply]
Maybe you could edit to spot it will be available via revive
